src/osd/OSD.cc

```diff
@@ -6264,23 +6265,42 @@ int OSD::probe_smart_device(const char *device, int timeout, std::string *result
 
   int ret = smartctl.spawn();
   if (ret != 0) {
-    derr << "failed run smartctl: " << smartctl.err() << dendl;
-    return ret;
+    std::string err_msg = "failed to run smartctl: " + smartctl.err();
+    derr << err_msg << dendl;
+    *result = err_msg;
+    return -1;
```

Comment:
I think that here we should return an error.  Pretty sure the spawn means we start the subprocess, and it's the join() where we get the final return value.